language/move-compiler/src/attr_derivation/async_deriver.rs

```diff
@@ -117,12 +131,23 @@ fn derive_module_for_async(
                     sign,
                     send_call,
                 ));
+
+                let mut struct_name = fun_def.name.to_string();
+                struct_name.replace_range(0..1, &struct_name[0..1].to_uppercase());
+                new_structs.push(new_struct(
```

Comment:
IIRC for every send message we will create a struct here. Am I understanding it correctly?

Reply:
Yes, the struct is annotated with #[event] so that the move-to-yul compiler can generate code to emit the corresponding event.